tap_recharge/client.py

```diff
@@ -52,51 +56,117 @@ class RechargeForbiddenError(RechargeError):
 class RechargeUnprocessableEntityError(RechargeError):
     pass
 
+class RechargeInvalidAPI(RechargeError):
+    pass
 
-class RechargeInternalServiceError(RechargeError):
+class RechargeInternalServiceError(Server5xxError):
     pass
 
+class RechargeUnimplementedResourceError(Server5xxError):
+    pass
 
-ERROR_CODE_EXCEPTION_MAPPING = {
-    400: RechargeBadRequestError,
-    401: RechargeUnauthorizedError,
-    402: RechargeRequestFailedError,
-    403: RechargeForbiddenError,
-    404: RechargeNotFoundError,
-    405: RechargeMethodNotAllowedError,
-    409: RechargeConflictError,
-    422: RechargeUnprocessableEntityError,
-    500: RechargeInternalServiceError}
+class RechargeThirdPartyServiceTimeoutError(Server5xxError):
+    pass
 
+# Doumentation: https://docs.rechargepayments.com/docs/api-standards#errors
+ERROR_CODE_EXCEPTION_MAPPING = {
+    400: {
+        "exception": RechargeBadRequestError,
+        "message": "The request was not understood by Recharge.",
+    },
+    401: {
+        "exception": RechargeUnauthorizedError,
+        "message": "The request was not able to be authenticated.",
+    },
+    402: {
+        "exception": RechargeRequestFailedError,
+        "message": "The request to the resource failed because of Payment issue.",
+    },
+    403: {
+        "exception": RechargeForbiddenError,
+        "message": "The request was authenticated but not authorized for the requested resource (Permission scope error).",
+    },
+    404: {
+        "exception": RechargeNotFoundError,
+        "message": "The requested resource was not found.",
+    },
+    405: {
+        "exception": RechargeMethodNotAllowedError,
+        "message": "The provided HTTP method is not supported by the URL.",
+    },
+    406: {
+        "exception": RechargeUnacceptableRequestError,
+        "message": "The request was unacceptable, or requesting a data source which is not allowed although permissions permit the request.",
+    },
+    409: {
+        "exception": RechargeConflictError,
+        "message": "The request is in conflict, or would create a conflict with an existing resource.",
+    },
+    415: {
+        "exception": RechargeJSONObjectError,
+        "message": "The request body was not a JSON object.",
+    },
+    422: {
+        "exception": RechargeUnprocessableEntityError,
+        "message": "The request was understood but cannot be processed due to invalid or missing supplemental information.",
+    },
+    426: {
+        "exception": RechargeInvalidAPI,
+        "message": "The request was made using an invalid API version.",
+    },
+    429: {
+        "exception": Server429Error,
```

Comment:
Replace `Server429Error` to `RechargeRateLimitError` for all occurences. Did we add backoff for 429 error?

Reply:
Yes, there is backoff for 429 error. Updated the **Server429Error** to **RechargeRateLimitError**